### PR TITLE
minimum rating filter and bookseller page added

### PR DESCRIPTION
**Pull request by ahabbaba1**
Allows users to view bestsellers in their own page, sort and filter results.

Added isBestseller to model to reflect new column in DB. Routing for bestsellers added. Used Sorter component with location passed in SortedCatalog to send correct DB query.

----------------------------

Users can filter books according to a minimum star rating. 

Added associations for Book and Review. Handled ratings fetch from DB for all existing calls. Added AntDesign star rater in Sorter with rating passed to SortedCatalog.

### CODE REVIEW

File: server/models/books.js [hunk 1]
```diff
@@ -36,10 +36,14 @@ const books = (sequelize, type) => {
         },
         publisher: {
             type: type.STRING(50)
+        },
```
Comment (paolaguitian) [inline server/models/books.js:39]: did you create a migration for this new field? I believe it wont just add to the db by adding to the file. A migration must be made or the DB must be deleted and inited again to have these changes! Double check this!

Comment (AquilaEternus) [reply server/models/books.js:39]: I just checked the DB. The field isBestseller is there, but the data in the DB is still intact. She must have found an alternative way of updating it without enforcing a sync(force: true) that would destroy and rebuild the columns and relations, but there are no new files or folders indicating that there was a migration

**Comment by AquilaEternus** [comment]
Profile management and book details still function properly on my end and no errors show up. Bestselling works as intended according to the author, and the select by book rating works when comparing the displayed results on screen to the data in the DB. Also, it was confirmed that the changes in the models as far as associations are concerned is required or else an error is thrown, and the isBestSeller column was added using a MySQL query seperate from sequelize.

File: server/models/books.js (hunk 1 above)
Comment (ahabbaba1) [reply server/models/books.js:39]: I manually created a column in Workbench with a SQL query and added associations that used existing foreign keys. I didn't find a migration or sync necessary.

**Pull request merged**